server/src/apub/community.rs

```diff
@@ -94,10 +104,85 @@ impl ActorType for Community {
     Ok(())
   }
 
+  fn send_delete(&self, creator: &User_, conn: &PgConnection) -> Result<(), Error> {
+    let group = self.to_apub(conn)?;
+    let id = format!("{}/delete/{}", self.actor_id, uuid::Uuid::new_v4());
+
+    let mut delete = Delete::default();
+    populate_object_props(&mut delete.object_props, &self.get_followers_url(), &id)?;
+
+    delete
+      .delete_props
+      .set_actor_xsd_any_uri(creator.actor_id.to_owned())?
+      .set_object_base_box(group)?;
+
+    // Insert the sent activity into the activity table
+    let activity_form = activity::ActivityForm {
+      user_id: self.creator_id,
+      data: serde_json::to_value(&delete)?,
+      local: true,
+      updated: None,
+    };
+    activity::Activity::create(&conn, &activity_form)?;
```

Comment:
I would just pass the parameters in and construct the ActivityForm struct inside the create() call, should make the code a good bit shorter.

Reply:
There's so many of these now, it makes sense to do this type of cleanup once we can make all this stuff more generic.